test/module/irohad/ametsuchi/flat_file_test.cpp

```diff
@@ -122,3 +127,62 @@ TEST_F(BlStore_Test, BlockStoreInitializationFromNonemptyFolder){
   // check that last ids of both block storages are the same
   ASSERT_EQ(bl_store1->last_id(), bl_store2->last_id());
 }
+
+TEST_F(BlStore_Test, EmptyDumpDir) {
+  auto res = check_consistency("");
+  ASSERT_EQ(res, nonstd::nullopt);
+}
+
+TEST_F(BlStore_Test, GetNonExistingFile) {
+  auto bl_store = FlatFile::create(block_store_path);
+  Identifier id = 98759385;  // random number that will not exist
+  auto res = bl_store->get(id);
+  ASSERT_EQ(res, nonstd::nullopt);
+}
+
+TEST_F(BlStore_Test, GetDirectory) {
+  auto bl_store = FlatFile::create(block_store_path);
+  ASSERT_EQ(bl_store->directory(), block_store_path);
+}
+
+TEST_F(BlStore_Test, GetDeniedBlock) {
+  std::vector<uint8_t> block(100000, 5);
+  auto bl_store = FlatFile::create(block_store_path);
+  auto id = 1u;
+  bl_store->add(id, block);
+
+  auto filename = boost::filesystem::path{block_store_path} / id_to_name(id);
+  chmod(filename.string().data(), 0);
+
+  auto res = bl_store->get(id);
+  ASSERT_EQ(res, nonstd::nullopt);
+}
+
+TEST_F(BlStore_Test, AddExistingId) {
+  std::vector<uint8_t> block(100000, 5);
+  auto bl_store = FlatFile::create(block_store_path);
+  auto id = 1u;
+  const auto file_name =
+      boost::filesystem::path{block_store_path} / id_to_name(id);
+  std::ofstream fout(file_name.string());
+  fout.close();
+
+  auto res = bl_store->add(id, block);
+  ASSERT_FALSE(res);
+}
+
+TEST_F(BlStore_Test, WriteEmptyFolder) {
+  std::vector<uint8_t> block(100000, 5);
+  auto bl_store = FlatFile::create("");
+  ASSERT_EQ(bl_store, nullptr);
+}
+
+TEST_F(BlStore_Test, WriteDeniedFolder) {
+  std::vector<uint8_t> block(100000, 5);
+  auto bl_store = FlatFile::create(block_store_path);
+  auto id = 1u;
+  chmod(block_store_path.data(), S_IRUSR | S_IXUSR);
```

Comment:
It should be possible to do without posix calls, but with use of `boost`

Reply:
Fixed